src/value/owned.rs

```diff
@@ -600,6 +590,16 @@ mod value_tests {
             (AsSequence, &|e| assert_eq!(None, e.as_sequence())),
             (AsStruct, &|e| assert_eq!(None, e.as_struct())),
         ];
+
+        // tests for element with annotations
+        let annotations = vec!["foo", "bar", "baz"];
+        let owned_elem = OwnedElement::new(
+            annotations.iter().map(|s| (*s).into()).collect(),
+            val.clone(),
+        );
+        let expected: Vec<String> = annotations.iter().map(|s| (*s).into()).collect();
+        assert_eq!(expected, annotation_strings(&owned_elem));
```

Comment:
I think I was trying to add 3 annotation to  each value in test cases and test those 3 annotations as you mentioned. But I think it would make more sense to have separate unit test for this. I will try to see if value lists can be used here.

Reply:
Here is a small, contrived example of mixing parameter cases and value lists.

```rust
#[cfg(test)]
mod test {
    use rstest::*;

    #[rstest]
    #[case::range_10_to_100(10, 100)]
    #[case::range_5_to_45(5, 45)]
    fn foo1(#[case] lval: i64, #[case] hval: i64, #[values(20, 30, 40)] other: i64) {
        assert!(lval < other);
        assert!(hval > other);
    }

    #[rstest(
        lval, hval,
        case::range_10_to_100(10, 100),
        case::range_5_to_45(5, 45),
        other => [20, 30, 40]
    )]
    fn foo2(lval: i64, hval: i64, other: i64) {
        assert!(lval < other);
        assert!(hval > other);
    }
}
```

The `foo2` example uses the macros I used, but apparently that is the [*old compact syntax*][1].  Apparently `rstest` 0.7 added this (my `ion-c-sys` tests used 0.6).  We don't need to port over now, but something for us to consider going forward.

[1]: https://docs.rs/rstest/0.7.0/rstest/attr.rstest.html#old-compact-syntax

Reply:
I do think having equivalence and non-equivalence classes of annotations as a value list could be used in the value test to test `Eq` more comprehensively, FWIW.

Reply:
See #158 (based on this PR) which makes some changes that illustrate what I was trying to articulate here.